config/reader.go

```diff
@@ -824,13 +824,16 @@ func CreateNewGRPCClient(ctx context.Context, cloudCfg *Cloud, logger golog.Logg
 		return nil, err
 	}
 
-	dialOpts := make([]rpc.DialOption, 1, 2)
-	dialOpts[0] = rpc.WithEntityCredentials(cloudCfg.ID,
-		rpc.Credentials{
-			Type:    rutils.CredentialsTypeRobotSecret,
-			Payload: cloudCfg.Secret,
-		},
-	)
+	dialOpts := make([]rpc.DialOption, 0, 2)
+	// Only add credentials when secret is set.
+	if cloudCfg.Secret != "" {
```

Comment:
@cheukt What's your thoughts on keeping this change in? It helps with local testing when no credentials are provided in the cloud config. Otherwise a empty string will be sent for the credentials.

Reply:
ya I'm good with it